pvlib/spectrum.py

```diff
@@ -0,0 +1,318 @@
+    mixed_coeff = _SPECTRL2_COEFFS['mixed_absorption'][:, np.newaxis]
+
+    # ET spectral irradiance correction for earth-sun distance seasonality
+    day_angle = 2 * np.pi * (dayofyear - 1) / 365.0  # Eq 2-3
+    earth_sun_distance_correction = (
+        1.000110
+        + 0.034221 * np.cos(day_angle)
+        + 0.001280 * np.sin(day_angle)
+        + 0.000719 * np.cos(2*day_angle)
+        + 0.000077 * np.sin(2*day_angle)
+    )  # Eq 2-2
+
+    # Rayleigh scattering
+    airmass = relative_airmass * surface_pressure / 101300
+    rayleigh_transmittance = np.exp(
+        # Note: the report uses 1.335 but spectrl2_2.c uses 1.3366
+        # -airmass / (wavelength**4 * (115.6406 - 1.335 / wavelength**2))
+        -airmass / (wavelength**4 * (115.6406 - 1.3366 / wavelength**2))
+    )  # Eq 2-4
+
+    # Aerosol scattering and absorption, Eq 2-6
+    aerosol_transmittance = np.exp(-optical_thickness * relative_airmass)
+
+    # Water vapor absorption, Eq 2-8
+    aWM = vapor_coeff * precipitable_water * relative_airmass
+    vapor_transmittance = np.exp(-0.2385 * aWM / (1 + 20.07 * aWM)**0.45)
+
+    # Ozone absorption
+    ozone_max_height = 22
+    h0_norm = ozone_max_height / 6370
+    ozone_mass_numerator = (1 + h0_norm)
+    ozone_mass_denominator = np.sqrt(cosd(apparent_zenith)**2 + 2 * h0_norm)
+    ozone_mass = ozone_mass_numerator / ozone_mass_denominator  # Eq 2-10
+    ozone_transmittance = np.exp(-ozone_coeff * ozone * ozone_mass)  # Eq 2-9
+
+    # Mixed gas absorption, Eq 2-11
+    aM = mixed_coeff * airmass
+    # Note: the report uses 118.93, but spectrl2_2.c uses 118.3
+    # mixed_transmittance = np.exp(-1.41 * aM / (1 + 118.93 * aM)**0.45)
+    mixed_transmittance = np.exp(-1.41 * aM / (1 + 118.3 * aM)**0.45)
+
+    # split out aerosol components for diffuse irradiance calcs
+    aerosol_scattering = (
+        np.exp(-scattering_albedo * optical_thickness * relative_airmass)
+    )  # Eq 3-9
+
+    aerosol_absorption = np.exp(
+        -(1 - scattering_albedo) * optical_thickness * relative_airmass
+    )  # Eq 3-10
+
+    return (
+        earth_sun_distance_correction,
+        rayleigh_transmittance,
+        aerosol_transmittance,
+        vapor_transmittance,
+        ozone_transmittance,
+        mixed_transmittance,
+        aerosol_scattering,
+        aerosol_absorption,
+    )
+
+
+def spectrl2(surface_tilt, apparent_zenith, aoi, ground_albedo,
+             surface_pressure, relative_airmass, precipitable_water, ozone,
+             dayofyear, aerosol_turbidity_500nm, scattering_albedo_400nm=0.945,
+             alpha=1.14, wavelength_variation_factor=0.095,
+             aerosol_asymmetry_factor=0.65):
+    """
+    Estimate spectral irradiance using the Bird Simple Spectral Model
+    (SPECTRL2).
+
+    The Bird Simple Spectral Model [1]_ produces terrestrial spectra between
+    0.3 and 4 um with a resolution of approximately 10 nm. Direct and diffuse
```

Comment:
Mild preference to use nm as the unit throughout, at least here in the docstring.  Not sure if nm is practical for the outputs and tests.